src/main/java/be/seeseemelk/mockbukkit/entity/EntityMock.java

```diff
@@ -376,9 +377,22 @@ public PermissionAttachment addAttachment(Plugin plugin, int ticks)
 	@Override
 	public void removeAttachment(PermissionAttachment attachment)
 	{
-		// TODO Auto-generated constructor stub
-		throw new UnimplementedOperationException();
+		if (attachment == null) {
+			throw new IllegalArgumentException("Attachment cannot be null");
+		}
+
+		if (permissionAttachments.contains(attachment)) {
+			permissionAttachments.remove(attachment);
+			PermissionRemovedExecutor ex = attachment.getRemovalCallback();
 
+			if (ex != null) {
+				ex.attachmentRemoved(attachment);
+			}
+
+			recalculatePermissions();
+		} else {
+			throw new IllegalArgumentException("Given attachment is not part of Permissible object " + this);
+		}
```

Comment:
Please use Allman braces (opening and closing brace always one a separate line).